src/main.rs

```diff
@@ -360,11 +363,10 @@ fn get_pt_connections(
     }
 }
 
-fn read_list_of_lists_vect32(filename: &str) -> Vec<Vec<i32>> {
-    let inpath = format!("serialised_data/{}.bin", filename);
-    let file = BufReader::new(File::open(inpath).unwrap());
-    let output: Vec<Vec<i32>> = bincode::deserialize_from(file).unwrap();
-    output
+fn deserialize_bincoded_file<T: DeserializeOwned>(filename: &str) -> T {
```

Comment:
This uses generics, https://doc.rust-lang.org/rust-by-example/generics.html, similar to Java and other languages. `T` is a type parameter that has to satisfy `DeserializeOwned` (something in serde)